server/src/main/java/org/opensearch/index/compositeindex/startree/StarTreeFieldSpec.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.startree;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Star tree index specific configuration
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class StarTreeFieldSpec implements ToXContent {
```

Comment:
This seems more of a configuration than specification. Shall we call this `StarTreeFieldConfiguration`?